File: Executables/AdminUi/packages/admin_api_types/lib/src/messages/message.dart
import 'package:json_annotation/json_annotation.dart';

part 'message.g.dart';

@JsonSerializable()
class Message {
  final String messageId;
  final String senderAddress;
  final String senderDevice;
  final DateTime sendDate;
  final int numberOfAttachments;
  final List<MessageRecipient> recipients;

  Message({
    required this.messageId,
    required this.senderAddress,
    required this.senderDevice,
    required this.sendDate,
    required this.numberOfAttachments,
    required this.recipients,
  });

  factory Message.fromJson(dynamic json) => _$MessageFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$MessageToJson(this);
}

@JsonSerializable()
class MessageRecipient {
  final String address;

  MessageRecipient({
    required this.address,
  });

  factory MessageRecipient.fromJson(dynamic json) => _$MessageRecipientFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$MessageRecipientToJson(this);
}

File: Sdks/AdminApi.Sdk/src/Endpoints/Messages/Types/Message.cs
namespace Backbone.AdminApi.Sdk.Endpoints.Messages.Types;

public class Message
{
    public string MessageId { get; set; } = null!;
    public string SenderAddress { get; set; } = null!;
    public string SenderDevice { get; set; } = null!;
    public DateTime SendDate { get; set; }
    public int NumberOfAttachments { get; set; }
    public List<MessageRecipient> Recipients { get; set; } = null!;
}
